pkg/cloud/scope/rosamachinepool.go

```diff
@@ -0,0 +1,183 @@
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to init MachinePool patch helper")
+	}
+
+	return &RosaMachinePoolScope{
+		Logger:                     *params.Logger,
+		Client:                     params.Client,
+		patchHelper:                ammpHelper,
+		capiMachinePoolPatchHelper: mpHelper,
+
+		Cluster:         params.Cluster,
+		ControlPlane:    params.ControlPlane,
+		RosaMachinePool: params.RosaMachinePool,
+		MachinePool:     params.MachinePool,
+		controllerName:  params.ControllerName,
+	}, nil
+}
+
+// RosaMachinePoolScope defines the basic context for an actuator to operate upon.
+type RosaMachinePoolScope struct {
+	logger.Logger
+	client.Client
+	patchHelper                *patch.Helper
+	capiMachinePoolPatchHelper *patch.Helper
+
+	Cluster         *clusterv1.Cluster
+	ControlPlane    *rosacontrolplanev1.ROSAControlPlane
+	RosaMachinePool *expinfrav1.ROSAMachinePool
+	MachinePool     *expclusterv1.MachinePool
+
+	controllerName string
+}
+
+// RosaMachinePoolName returns the rosa machine pool name.
+func (s *RosaMachinePoolScope) RosaMachinePoolName() string {
+	return s.RosaMachinePool.Name
+}
+
+// NodePoolName returns the nodePool name of this machine pool.
+func (s *RosaMachinePoolScope) NodePoolName() string {
+	return s.RosaMachinePool.Spec.NodePoolName
+}
+
+// ClusterName returns the cluster name.
+func (s *RosaMachinePoolScope) RosaClusterName() string {
+	return s.ControlPlane.Spec.RosaClusterName
+}
+
+// ControlPlaneSubnets returns the control plane subnets.
+func (s *RosaMachinePoolScope) ControlPlaneSubnets() []string {
+	return s.ControlPlane.Spec.Subnets
+}
+
+// InfraCluster returns the AWS infrastructure cluster or control plane object.
+func (s *RosaMachinePoolScope) InfraCluster() cloud.ClusterObject {
+	return s.ControlPlane
+}
+
+// ClusterObj returns the cluster object.
+func (s *RosaMachinePoolScope) ClusterObj() cloud.ClusterObject {
+	return s.Cluster
+}
+
+// ControllerName returns the name of the controller that
+// created the RosaMachinePool.
+func (s *RosaMachinePoolScope) ControllerName() string {
+	return s.controllerName
+}
+
+func (s *RosaMachinePoolScope) GetSetter() conditions.Setter {
+	return s.RosaMachinePool
+}
+
+// RosaMchinePoolReadyFalse marks the ready condition false using warning if error isn't
+// empty.
+func (s *RosaMachinePoolScope) RosaMchinePoolReadyFalse(reason string, err string) error {
+	severity := clusterv1.ConditionSeverityWarning
+	if err == "" {
+		severity = clusterv1.ConditionSeverityInfo
+	}
+	conditions.MarkFalse(
+		s.RosaMachinePool,
+		expinfrav1.RosaMachinePoolReadyCondition,
+		reason,
+		severity,
+		err,
+	)
+	if err := s.PatchObject(); err != nil {
+		return errors.Wrap(err, "failed to mark rosa machinepool not ready")
+	}
+	return nil
+}
+
+// PatchObject persists the control plane configuration and status.
+func (s *RosaMachinePoolScope) PatchObject() error {
+	return s.patchHelper.Patch(
+		context.TODO(),
```

Comment:
We should not write new code that does this - thread the context please.

Reply:
what do you mean by new code?

Reply:
I think why no body cared to thread this, is that controller-runetime ensures that only 1 thread is handling an object reconciliation at a time.
But yeah I see your point, I will add a `context` to this func.